### PR TITLE
chore: consistent telemetry events for prereqs check

### PR DESCRIPTION
**Pull request by a1exwang**
image

### CODE REVIEW

File: packages/vscode-extension/src/debug/prerequisitesHandler.ts [hunk 1]
```diff
@@ -363,85 +387,83 @@ export async function checkAndInstall(): Promise<Result<any, FxError>> {
...
-    });
+    ctx.properties[TelemetryProperty.DebugCheckResults] = JSON.stringify(checkResults);
     return err(fxError);
   }
-
-  return ok(null);
+  return ok(undefined);
 }
 
 async function checkM365Account(prefix: string, showLoginPage: boolean): Promise<CheckResult> {
```
Comment (swatDong) [inline packages/vscode-extension/src/debug/prerequisitesHandler.ts:400]: Is `async` still needed?

File: packages/vscode-extension/src/debug/prerequisitesHandler.ts [hunk 2]
```diff
@@ -196,14 +209,19 @@ export async function checkPrerequisitesForGetStarted(): Promise<Result<any, FxE
   return ok(null);
 }
 
-export async function checkAndInstall(): Promise<Result<any, FxError>> {
+export async function checkAndInstall(): Promise<Result<void, FxError>> {
+  const projectComponents = await commonUtils.getProjectComponents();
+  return localTelemetryReporter.runWithTelemetryProperties(
```
Comment (swatDong) [inline packages/vscode-extension/src/debug/prerequisitesHandler.ts:214]: await?

Comment (a1exwang) [reply packages/vscode-extension/src/debug/prerequisitesHandler.ts]: Added.

File: packages/vscode-extension/src/debug/prerequisitesHandler.ts (hunk 1 above)
Comment (a1exwang) [reply packages/vscode-extension/src/debug/prerequisitesHandler.ts]: removed

File: packages/vscode-extension/src/debug/prerequisitesHandler.ts [hunk 3]
```diff
@@ -128,34 +126,49 @@ const ProgressMessage: { [key: string]: string } = Object.freeze({
...
 
+async function runWithCheckResultTelemetry(
+  eventName: string,
+  action: (ctx: TelemetryContext) => Promise<CheckResult>
+): Promise<CheckResult> {
+  return await localTelemetryReporter.runWithTelemetryGeneric(
+    eventName,
+    action,
+    (result: CheckResult) => {
+      return result.result == ResultStatus.success ? undefined : result.error;
```
Comment (kuojianlu) [inline packages/vscode-extension/src/debug/prerequisitesHandler.ts:137]:
```suggestion
      return result.result === ResultStatus.success ? undefined : result.error;
```

**Review by kuojianlu — APPROVED** [review]

**Review by tecton — APPROVED** [review]

**Pull request merged**